terraso_backend/apps/core/views.py

```diff
@@ -23,3 +40,130 @@ def check_db_access():
     Landscape.objects.count()
     Group.objects.count()
     User.objects.count()
+
+
+@staff_member_required
+def create_restore_job(request):
+    if request.method != "POST":
+        return HttpResponseNotAllowed(permitted_methods=["POST"])
+
+    with transaction.atomic():
+        cursor = get_connection().cursor()
+        cursor.execute(f"LOCK TABLE {BackgroundTask._meta.db_table}")
+        task = BackgroundTask(created_by=request.user, status="running")
+        try:
+            active_tasks = BackgroundTask.objects.filter(status="running").count()
+            if active_tasks:
+                return HttpResponse(
+                    json.dumps({"message": "Already a restore task running"}),
+                    "application/json",
+                    400,
+                )
+            task.save()
+        except Exception:
+            task.delete()
+        finally:
+            cursor.close()
+        thread = threading.Thread(
+            target=restore, args=[task, request.user.id, request.session.session_key], daemon=True
+        )
+        thread.start()
+
+    return HttpResponse(json.dumps({"taskId": task.id}), "application/json")
+
+
+def check_restore_job_status(request, task_id):
+    if request.method != "GET":
+        return HttpResponseNotAllowed(permitted_methods=["GET"])
+    try:
+        task = BackgroundTask.objects.get(id=task_id)
+        session = Session.objects.get(session_key=request.session.session_key)
+        session_user_id = session.get_decoded().get("_auth_user_id")
+        if session_user_id != str(task.created_by_id):
+            return HttpResponseNotFound(json.dumps({"message": "Restricted"}), "application/json")
+    except BackgroundTask.DoesNotExist | Session.DoesNotExist:
+        return HttpResponseNotFound(
+            json.dumps({"message": f"No task with id {task_id}"}), "application/json"
+        )
+    return HttpResponse(json.dumps({"status": task.status}), "application/json")
+
+
+def _sync_s3_buckets(
+    source_buckets: list[str], dest_buckets: list[str], aws_credentials: tuple[str, str, str]
+):
+    """Sync contents of source_buckets to dest_buckets."""
+    env = {
+        key: value
+        for key, value in zip(
+            ["AWS_ACCESS_KEY", "AWS_SECRET_ACCESS_KEY", "AWS_DEFAULT_REGION"], aws_credentials
+        )
+    }
+    env.update(os.environ)
+    for source, dest in zip(source_buckets, dest_buckets):
+        subprocess.run(["aws", "s3", "sync", "s3://" + source, "s3://" + dest], env=env)
+
+
+JOB_WAIT_TIME_SEC = 5
```

Comment:
suggestion: move this to the top of the file and give it a more descriptive name.